BinaryEncoding.md

```diff
@@ -424,12 +426,13 @@ a `data_segment` is:
 
 Custom section `name` field: `"name"`
 
-The names section is a [custom section](#high-level-structure). 
+The name section is a [custom section](#high-level-structure). 
 It is therefore encoded with id `0` followed by the name string `"name"`.
-Like all custom sections, a validation error in this section does not cause validation of the module to fail.
+Like all custom sections, a validation error in this section does not cause validation of the module to fail. Furthermore, the wasm implementation may choose to keep and use any valid information it gathered from this section up to the point of invalidation, or discard it. The wasm implementation may also choose to read and validate this section lazily, after the module has been instantiated, should debugging be required.
```

Comment:
Here, too, I'd avoid speaking of "validation" and "validation errors" wrt to the custom data. Also, I think you can shorten the latter part to say sth like "It's up to the implementation how it handles a malformed or partially malformed name section."